ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vlan_interfaces.py

```diff
@@ -104,8 +104,13 @@ def _check_virtual_router_mac_address(vlan_interface_config: dict, variables: li
 
         # Only set Anycast v6 GW if VARPv6 is not set
         if vlan_interface_config.get("ip_virtual_router_addresses") is None:
-            vlan_interface_config["ipv6_address_virtual"] = svi.get("ipv6_address_virtual")
-            vlan_interface_config["ipv6_address_virtuals"] = svi.get("ipv6_address_virtuals")
+            if (ipv6_address_virtual := svi.get("ipv6_address_virtual")) is not None:
+                # The singular ipv6_address_virtual is deprecated from eos_cli_config_gen. So set as list item into the new key.
+                vlan_interface_config["ipv6_address_virtuals"] = [ipv6_address_virtual]
+
+            if (ipv6_address_virtuals := svi.get("ipv6_address_virtuals")) is not None:
+                vlan_interface_config.setdefault("ipv6_address_virtuals", []).extend(ipv6_address_virtuals)
+
             _check_virtual_router_mac_address(vlan_interface_config, ["ipv6_address_virtual", "ipv6_address_virtuals"])
```

Comment:
Given that we don't set `ipv6_address_virtual` key anymore in vlan_interface_config I think we don't need it in this line anymore

Reply:
fixed